web/src/app/pages/main-page-container/main-page/secondary-side-panel/submission-panel/submission-panel.component.ts

```diff
@@ -54,6 +57,17 @@ export class SubmissionPanelComponent implements OnInit, OnDestroy {
           this.tasks = submission.job
             ?.getTasksSorted()
             .filter(task => !task.addLoiTask);
+          // Get image URL upon initialization to not send firebase requests multiple times
```

Comment:
Nit 1: Can we refactor this into a separate function?

Nit 2: s/firebase/Firebase`